juju/juju
# Preliminary FAN support

## Description of change
This PR introduces basic support for FAN networking in containers
A new fan-config model option is introduced, it's syntax is "underlay1:overlay1;underlay2:overlay2", e.g. 172.31.0.0/16:253.0.0.0/8. A new fanconfigurer worker in the machine context is setting up fan on machines if the model has fan-config set.
Currently, if fan-config is set -all- containers created in the model will have fan IP.
Please remember that this is still a WiP.

## QA steps
- Bootstrap a model on AWS with model-config fan-config="173.31.0.0/16:253.0.0.0/8" (use the proper subnet for VPC you're using)
- juju add-machine lxd -n 2
- connect to one of the host machines, enter the container and check connectivity to the container on the other machine


## Documentation changes
This change will need a separate doc on container networking setup.

## Bug reference
